### PR TITLE
Persistence support for replication state for execution

### PR DESCRIPTION
**Pull request by samarabbas**
Added support for workflow mutable state to create/read/update
replication state for an execution.

Support for transfer task creation for Replication Task.

Added unit test for replication state for execution

### CODE REVIEW

File: common/persistence/dataInterfaces.go [hunk 1]
```diff
@@ -164,6 +165,15 @@ type (
 		ClientImpl                   string
 	}
 
+	// ReplicationState represents
```
Comment (wxing1292) [inline common/persistence/dataInterfaces.go]: typo

File: schema/cadence/versioned/v0.5/history_replication_task.cql [hunk 2]
```diff
@@ -0,0 +1,22 @@
+-- Replication information for each cluster
```
Comment (wxing1292) [inline schema/cadence/versioned/v0.5/history_replication_task.cql]: plz change the version from 0.5 to 0.6

File: schema/cadence/schema.cql [hunk 3]
```diff
@@ -60,8 +75,12 @@ CREATE TYPE transfer_task (
   target_run_id              uuid,   -- The external run ID that this transfer task is doing work for.
   target_child_workflow_only boolean, -- The whether target child workflow only.
   task_list                  text,
-  type                       int,    -- enum TaskType {ActivityTask, DecisionTask, DeleteExecution, CancelExecution, StartChildExecution}
+  type                       int,    -- enum TaskType {ActivityTask, DecisionTask, DeleteExecution, CancelExecution, StartChildExecution, ReplicationTask}
   schedule_id                bigint,
+  first_event_id             bigint,
```
Comment (wxing1292) [inline schema/cadence/schema.cql]: could you add some description here?

File: common/persistence/dataInterfaces.go [hunk 4]
```diff
@@ -304,6 +318,22 @@ type (
 		EventID             int64
 	}
 
+	// ReplicationTask is the transfer task created for shipping history replication events to other clusters
+	ReplicationTask struct {
+		TaskID              int64
+		DomainID            string
```
Comment (wxing1292) [inline common/persistence/dataInterfaces.go]: is this DomainID necessary?

File: common/persistence/dataInterfaces.go (hunk 1 above)
Comment (samarabbas) [reply common/persistence/dataInterfaces.go]: done.

File: schema/cadence/versioned/v0.5/history_replication_task.cql (hunk 2 above)
Comment (samarabbas) [reply schema/cadence/versioned/v0.5/history_replication_task.cql]: done.

File: schema/cadence/schema.cql (hunk 3 above)
Comment (samarabbas) [reply schema/cadence/schema.cql]: done.

File: common/persistence/dataInterfaces.go (hunk 4 above)
Comment (samarabbas) [reply common/persistence/dataInterfaces.go]: You are right.  Replication task is always going to be for the same domain.  Removed.

**Pull request merged**